### ISSUE TITLE
Support Python 3.10

### ISSUE THREAD

**Issue by decaz** [PyYoshi/cChardet#77]
... and prepare for Python 3.11 (dev).

**Comment by spenpal2000** [comment]
Any news on making this library Python 3.10 supported?

**Comment by oleksandr-kuzmenko** [comment]
https://github.com/PyYoshi/cChardet/pull/78

**Comment by oleksandr-kuzmenko** [comment]
JFYI https://github.com/PyYoshi/cChardet/pull/78

**Comment by ooliver1** [comment]
After 6 months, will that PR get merged or even looked at

**Comment by MikBighne98** [comment]
Future updates like python 3.10 support coming or the project is dropped?

**Comment by ooliver1** [comment]
it seems as though cchardet has been abandoned, yet it is depended on by many large projects

**Comment by NebularNerd** [comment]
Are there any good alternatives to cchardet? If the repo is not getting any love then quite a few projects will need a replacement.

**Comment by ooliver1** [comment]
^, it is possible to install `gcc` but this wont be the biggest issue forever, 3.11/12 may somehow break this

**Comment by NebularNerd** [comment]
> ^, it is possible to install `gcc` but this wont be the biggest issue forever, 3.11/12 may somehow break this

I assume this is for *nix users, I'm on Windows and it keeps throwing up the 'C++ 14 Required' error when I try to install. I assume because for Windows it's trying to compile using C++ instead of gcc

Does anyone know if I can manually compile this on Windows using my MinGW gcc install? I'd rather not download multi GB's of Visual Studio just for one python package.

**Comment by ooliver1** [comment]
> > ^, it is possible to install `gcc` but this wont be the biggest issue forever, 3.11/12 may somehow break this
> 
> I assume this is for *nix users, I'm on Windows and it keeps throwing up the 'C++ 14 Required' error when I try to install. I assume because for Windows it's trying to compile using C++ instead of gcc
> 
> Does anyone know if I can manually compile this on Windows using my MinGW gcc install? I'd rather not download multi GB's of Visual Studio just for one python package.

@NebularNerd yes this was *nix, here are steps i found for windows using mingw
- add `C:\MinGW\bin` to PATH
- edit `PYTHONPATH\Lib\distutils` with a `distutils.cfg` file containing
```toml
[build]
compiler=mingw32
```
<https://stackoverflow.com/a/5051281>

**Comment by banagale** [comment]
> Are there any good alternatives to cchardet? If the repo is not getting any love then quite a few projects will need a replacement.

It depends on what you're trying to do. There's an `MIT` licensed  package called [`charset_normalizer`](https://github.com/Ousret/charset_normalizer) many seem to have switched to.

`charset_normalizer` focuses on providing you the actual text content in usable, unicode form. 

Whereas, it seems like cchardet focuses on trying to tell you what a text file is encoded in. In a project I'm working on, this detected encoding is attempted to be used with an `open()`.

`charset_normalizer` is like, "why?" figures out the most likely original encoding scheme and lets you immediately get the text content.

If you look, it is specifically compared with this package and calls out this package, `cChardet`'s apparent use of a cpp binding. It also claims it has higher accuracy but possibly less speed.

**Comment by mohd-akram** [comment]
It was working for me on Python 3.10, but now fails to install on Python 3.11:

```
/usr/bin/clang -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -Isrc/ext/uchardet/src -I/opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c src/cchardet/_cchardet.cpp -o build/temp.macosx-12.0-x86_64-cpython-311/src/cchardet/_cchardet.o
src/cchardet/_cchardet.cpp:196:12: fatal error: 'longintrepr.h' file not found
  #include "longintrepr.h"
           ^~~~~~~~~~~~~~~
1 error generated.
error: command '/usr/bin/clang' failed with exit code 1
```

**Comment by SimplicityGuy** [comment]
There are 2 PRs #78 and #80 that will address this. @PyYoshi, can you merge and release a new build, please?

**Comment by ooliver1** [comment]
> There are 2 PRs #78 and #80 that will address this. @PyYoshi, can you merge and release a new build, please?

It is pretty established they have abandoned cchardet, see the PRs you referenced, #78 is nearly 1 year old.

**Comment by SimplicityGuy** [comment]
> It is pretty established they have abandoned cchardet, see the PRs you referenced, #78 is nearly 1 year old.

Indeed. It's unfortunate since right now many downstream dependencies can't be completely installed with Python 3.11 due to build issues.

**Comment by NebularNerd** [comment]
At this stage it's come down to either moving to [charset_normalizer](https://github.com/Ousret/charset_normalizer) or if someone is willing to, fork this and make cchardet-ng or similar.
